iron-icon.html

```diff
@@ -160,9 +164,15 @@
             this._iconset = this._meta.byKey(this._iconsetName);
             if (this._iconset) {
               this._iconset.applyIcon(this, this._iconName, this.theme);
+              this.unlisten(window, 'iron-iconset-added', '_updateIcon');
+              clearTimeout(this._warningTimeout);
             } else {
-              this._warn(this._logf('_updateIcon', 'could not find iconset `'
-                + this._iconsetName + '`, did you import the iconset?'));
+              this.listen(window, 'iron-iconset-added', '_updateIcon');
+
+              this._warningTimeout = setTimeout(function() {
+                this._warn(this._logf('_updateIcon', 'could not find iconset `'
+                    + this._iconsetName + '`, did you import the iconset?'));
+              }.bind(this), 5000);
```

Comment:
This is true. It also does make icons more expensive when the iconset isn't loaded yet, and you can have large numbers of icons on the page at once.

Removed for now and can revisit if people seem to run into iconset loading issues.